config.go

```diff
@@ -140,6 +141,15 @@ func (b *Builder) fixupConfig(sys *types.SystemContext) {
 		b.SetArchitecture(ps.Architecture)
 		b.SetVariant(ps.Variant)
 	}
+	if b.Variant() == "" {
+		if sys != nil && sys.VariantChoice != "" {
+			b.SetVariant(sys.VariantChoice)
+		}
+		// in case the arch string we started with was shorthand for a known arch+variant pair, normalize it
+		_, normalizedArch, normalizedVariant := libimage.NormalizePlatform(b.OS(), b.Architecture(), b.Variant())
```

Comment:
Using `libimage.NormalizePlatform` instead of directly calling `platforms.Normalize` cause https://github.com/containerd/containerd/pull/2417 made a change which will return empty string for certain variant and to resolve correct variant we must use `platforms.Parse` which `NormalizePlatform` already uses so instead use that.

Reply:
Why do we want to deviate from what `platforms.Normalize()` is doing here?

Reply:
I think we must use `platforms.Parse` after generating output from `platforms.Normalize` since `platforms.Normalize` return empty value for certain variants for example consider a case of `arm64` and variant `v8`, but `platforms.Parse` sets the correct value after reading this empty value. I think change went here: https://github.com/containerd/containerd/pull/2417

Since `libimage.NormalizePlatform` was already using `platforms.Parse` hence i decided to re-use it instead of calling `platform.Normalize` and then `platform.Parse(`.

Reply:
Then should we be using `libimage.NormalizePlatform()` in other places that we currently call `platforms.Normalize()`?

Reply:
Yes I agree, I was trying for this change in the same PR but it looked out of scope. Should we do it in a new PR ?

Reply:
I don't see the benefit of creating an internal inconsistency that we know we'll have to come back and fix later when we're well-situated to just sort it all out up front.

Reply:
Added a new commit which does this change and a test was retrofitted which sets `variant` without any `--variant` flag to the image.